src/OpenTelemetry.AutoInstrumentation.Native/rejit_handler.cpp

```diff
@@ -370,35 +288,41 @@ void RejitHandler::AddNGenModule(ModuleID moduleId)
     RequestRejitForInlinersInModule(moduleId);
 }
 
-void RejitHandler::EnqueueForRejit(std::vector<ModuleID>& modulesVector, std::vector<mdMethodDef>& modulesMethodDef)
+void RejitHandler::RequestRejit(std::vector<ModuleID>& modulesVector, std::vector<mdMethodDef>& modulesMethodDef)
 {
     const size_t length = modulesMethodDef.size();
 
-    auto moduleIds = new ModuleID[length];
-    std::copy(modulesVector.begin(), modulesVector.end(), moduleIds);
-
-    auto mDefs = new mdMethodDef[length];
-    std::copy(modulesMethodDef.begin(), modulesMethodDef.end(), mDefs);
-
     // Create module and methods metadata.
     for (size_t i = 0; i < length; i++)
     {
-        GetOrAddModule(moduleIds[i])->GetOrAddMethod(mDefs[i]);
+        GetOrAddModule(modulesVector[i])->GetOrAddMethod(modulesMethodDef[i]);
     }
 
-    // Enqueue rejit
-    m_rejit_queue->push(std::make_unique<RejitItem>((int) length, std::unique_ptr<ModuleID>(moduleIds),
-                                                    std::unique_ptr<mdMethodDef>(mDefs)));
-}
+    HRESULT hr;
+    auto profilerInfo = m_profilerInfo;
+    auto profilerInfo10 = m_profilerInfo10;
 
-void RejitHandler::Shutdown()
-{
-    m_rejit_queue->push(RejitItem::CreateEndRejitThread());
-    if (m_rejit_queue_thread->joinable())
+    if (profilerInfo10 != nullptr)
```

Comment:
What is the reason to have this condition here?
`m_profilerInfo` is `ICorProfilerInfo4` which is interface containing `RequestReJIT` fuction https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/icorprofilerinfo4-requestrejit-method.

Do you want to use here `RequestReJITWithInliners ` https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/icorprofilerinfo10-requestrejitwithinliners-method or you have some other reason?

Reply:
Good point @Kielek, ICorProfilerInfo4 is .NET Framework 4.5 so we don't need to worry about that version anymore, we can consider that we have ICorProfilerInfo6 and above. Cleaning it up...

Reply:
We can remove this multiple profiler interface members but it is better to do in a separate PR. I will do it as soon as this one is merged.